enhancements/installer/agent-late-binding.md

```diff
@@ -0,0 +1,179 @@
+
+### Non-Goals
+
+- Once a host boots from disk, booting the Discovery ISO again to return it to
+  the collection of free Agents is considered to be a manual step.  There are
+  several options that we will explore whereby this can be done automatically,
+  but that discussion is out of scope for this enhancement.
+- At a later stage, Agents may be automatically assigned to clusters based on
+  their discovered inventory and/or network configurations. This is future work
+  and not in the scope of this enhancement.
+
+## Proposal
+
+The first major change is related to the Discovery ISO.  The only API change
+(which is non-breaking) is that ClusterRef will no longer mandatory.  Once an
+agent registers with the service, its corresponding Agent CR must be created in
+the namespace of the InfraEnv which generated its Discovery ISO.  The
+controller currently relies on the fact that an agent is always associated with
+a cluster, and an InfraEnv is always associated with a cluster, to associate
+Agents with InfraEnvs.  Now that these associations are optional, the
+InfraEnv's identity must be encoded in the ISO, and the agent must pass it to
+the service upon registration.  When an agent registers, the backend will
+create the Agent CR in the InfraEnv's namespace, and if the InfraEnv references
+a ClusterDeployment, the Agent will automatically be associated with that
+ClusterDeployment.
+
+The second major change is to the Agent CRD, where the ClusterRef is no longer
+mandatory.  This association may now be deleted or changed dynamically, unless
+the Agent is part of an ongoing cluster installation, in which case the
+installation must first be cancelled.  If the host had started/completed its
+installation, a new Agent Condition would indicate that the user needs to boot
+the host into the Discovery ISO once again.
+
+Agent validations that do not depend on being part of a cluster will run for
+Agents that are both associated with clusters and those that are not.  Examples
+of these include minimum hardware for any role, synced NTP, and registry
+access.  Validations that do depend on being part of a cluster, such as
+connectivity checks, will run only for Agents associated with clusters.
+
+### User Stories
+
+#### Story 1
+
+As an Infrastructure Admin, I want to create a Discovery ISO and use it to boot
+a pool of hosts for use by Cluster Creators to install OpenShift on.
+
+#### Story 2
+
+As an Infrastructure Admin, I want to add Discovery ISOs to hosts as a boot
+option, and then ship the hosts to remote locations where they will be booted
+at a later time for use by Cluster Creators to install OpenShift on.
+
+#### Story 3
+
+As a Cluster Creator, I want to view a collection of available Agents and use
+them for OpenShift cluster creation or expansion.
+
+#### Story 4
+
+As a Cluster Creator, I want to reassign an Agent from one OpenShift cluster to
+another.
+
+### Implementation Details/Notes/Constraints [optional]
```

Comment:
This might be a good place to detail the API changes and show examples. Or maybe under "Design Details".